core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -167,14 +166,37 @@ private[kafka] object ConsumerStage {
         )
       }
 
-      val newStages = stages.get(key.groupId) match {
+      stages.get(key.groupId) match {
         case Some(s) =>
           require(s == stage, s"CommittableOffset [$committableOffset] origin stage must be same as other " +
             s"stage with same groupId. Expected [$s], got [$stage]")
           stages
         case None =>
           stages.updated(key.groupId, stage)
       }
+    }
+
+    override def updated(committableOffset: CommittableOffset): CommittableOffsetBatch = {
+      val partitionOffset = committableOffset.partitionOffset
+      val key = partitionOffset.key
+
+      val newOffsets = offsets.updated(key, committableOffset.partitionOffset.offset)
+
+      new CommittableOffsetBatchImpl(newOffsets, updatedStages(stages, key, committableOffset))
+    }
+
+    override def updated(committableOffsets: Seq[CommittableOffset]): CommittableOffsetBatch = {
+      val (newStages, newOffsets) = committableOffsets.groupBy(_.partitionOffset.key).foldLeft((stages, offsets)) {
+        case ((aggStages, aggOffsets), (key, committableOffsetsForKey)) =>
+          val commitToUse = committableOffsetsForKey.sortBy(_.partitionOffset.offset).head
+          val partitionOffset = commitToUse.partitionOffset
+          val key = partitionOffset.key
+          val offset = commitToUse.partitionOffset.offset
+
+          val offsetToUpdateTo = Math.max(aggOffsets.getOrElse(key, offset), offset)
```

Comment:
but here it's only using that offset if it's greater than existing in the batch
I'm not sure I understand what this method is intended to do.